x-pack/plugins/observability/server/utils/unwrap_es_response.ts

```diff
@@ -4,11 +4,45 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
+import { ElasticsearchClientError, ResponseError } from '@elastic/elasticsearch/lib/errors';
 import type { UnwrapPromise } from '@kbn/utility-types';
+import { inspect } from 'util';
+
+export class WrappedElasticsearchClientError extends Error {
+  originalError: ElasticsearchClientError;
+  constructor(originalError: ElasticsearchClientError) {
+    super(originalError.message);
+
+    const stack = this.stack;
+
+    this.originalError = originalError;
+
+    if (originalError instanceof ResponseError) {
+      // make sure ES response body is visible when logged to the console
+      // @ts-expect-error
+      this.stack = {
+        valueOf() {
+          const value = stack?.valueOf() ?? '';
+          return value;
+        },
+        toString() {
+          const value =
+            stack?.toString() +
+            `\nResponse: ${inspect(originalError.meta.body, { depth: null })}\n`;
+          return value;
+        },
+      };
+    }
+  }
+}
 
 export function unwrapEsResponse<T extends Promise<{ body: any }>>(
   responsePromise: T
 ): Promise<UnwrapPromise<T>['body']> {
-  return responsePromise.then((res) => res.body);
+  return responsePromise
+    .then((res) => res.body)
+    .catch((err) => {
+      // make sure stacktrace is relative to where client was called
+      throw new WrappedElasticsearchClientError(err);
```

Comment:
Not sure you can use it for anything in this context but have you played around with `Error.captureStackTrace` ?

Reply:
I did, but that won't allow me to change the output of the error in the console. But I'll talk to some ops folks to see if we can address that issue for all errors rather than this hack.